o4/o4.py

```diff
@@ -866,8 +894,11 @@ def man(cmd):
     run_cmd(cmd)
 
     actual_cl, _ = get_fstat_cache(changelist)
-    with open('.o4/changelist', 'wt') as fout:
-        print(actual_cl, file=fout)
+    try:
+        os.remove(INCOMPLETE_INDICATOR)
+    except FileNotFoundError:
+        with open(SYNCED_CL_FILE, 'wt') as fout:
+            print(actual_cl, file=fout)
```

Comment:
That looks like an anti-pattern - control-flow by exception. Is there a specific reason you are doing `os.path.exists` earlier up in the file but not in this location?

Also, the file is different, i.e. if you fail to delete `INCOMPLETE_INDICATOR` you are just opening `SYNCED_CL_FILE`.  Is it guaranteed to exist at this point?

Reply:
The SYNCED file is being created if there was no INCOMPLETE file. The only purpose of INCOMPLETE is to prevent SYNCED (which says "the sync to the given changelist is true and complete") from being created. This is communication among several concurrent processes.